src/Controller/Produtos/ProdutoApiController.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Controller\Produtos;
+
+use App\Repository\ProdutoRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+class ProdutoApiController extends AbstractController
+{
+    #[Route('/api/produtos', name: 'api_produtos', methods: ['GET'])]
+    public function listarProdutos(Request $request, ProdutoRepository $produtoRepository): JsonResponse
+    {
+        $nome = $request->query->get('nome');
+        
+        $produtos = $produtoRepository->buscarProdutosAtivos($nome);
+
+        $dados = array_map(fn($produto) => [
+            'nome' => $produto->getNome(),
+            'quantidade' => $produto->getQuantidade(),
+            'valorUnitario' => $produto->getValorUnitario(),
+        ], $produtos);
+
+        return $this->json($dados);
+    }

```

Comment:
Tem uma outra forma massa de fazer também, tu implementa um interface na entidade de Produto chamada JsonSerialize, ai ela vai te obrigar criar um método (a interface é como se fosse um contrato que tu tem que cumprir), ai lá você especifica os campos...

```php
puiblic function jsonSerializer(): array
{
    return [
            'nome' => $this->getNome(),
            'quantidade' => $this->getQuantidade(),
            'valorUnitario' => $this->getValorUnitario(),
    ];
}
```

ai no Controller ficaria assim, igual ensinei na sala segunda-feira...

@rafaelyudji se não consegui me chama

```suggestion
    #[Route('/api/produtos', name: 'api_produtos', methods: ['GET'])]
    public function listarProdutos(Request $request, ProdutoRepository $produtoRepository): JsonResponse
    {
        $nome = $request->query->get('nome');
        
        $produtos = $produtoRepository->buscarProdutosAtivos($nome);

        return $this->json($produtos);
    }

```